dw-ripple.js

```diff
@@ -58,11 +49,65 @@ export class DwRipple extends Ripple {
     }
   }
 
-  constructor(){
+  constructor() {
     super();
-    this.primary=false;
+    this.primary = false;
     this.secondary = false;
     this.disabled = false;
+    this.__init();
+  }
+
+  __init() {
+    this._rippleHander = new RippleHandlers(async () => this);
+    const parent = this.parentNode;
+    this.__onMouseDown = this.__onMouseDown.bind(this);
+    this.__onTouchStart = this.__onTouchStart.bind(this);
+    this.__onMouseUp = this.__onMouseUp.bind(this);
+    this.__onTouchEnd = this.__onTouchEnd.bind(this);
+
+    parent.addEventListener('mouseenter', this._rippleHander.startHover);
+    parent.addEventListener('mouseleave', this._rippleHander.endHover);
+    parent.addEventListener('mousedown', this.__onMouseDown);
+    parent.addEventListener('touchstart', this.__onTouchStart);
+    parent.addEventListener('focus', this._rippleHander.startFocus);
+    parent.addEventListener('blur', this._rippleHander.endFocus);
+  }
+
+  __onMouseDown(e) {
+    window.addEventListener('mouseup', this.__onMouseUp);
+    this._rippleHander.startPress(e);
+  }
+
+  __onTouchStart(e) {
+    window.addEventListener('touchend', this.__onTouchEnd);
+    this._rippleHander.startPress(e);
+  }
+
+  __onMouseUp() {
+    window.removeEventListener('mouseup', this.__onMouseUp);
+    this._rippleHander.endPress();
+  }
+
+  __onTouchEnd() {
+    window.removeEventListener('touchend', this.__onTouchEnd);
+    this._rippleHander.endPress();
+  }
+
+  disconnectedCallback() {
+    const parent = this.parentNode;
+    console.log('disconne', parent)
```

Comment:
@nirmalbaldaniya-dream Please remove `console.log`

Reply:
@nirmalbaldaniya-dream This isn't yet removed.